File: Baekjoon/구현/2346.py
# https://www.acmicpc.net/problem/2346

n = int(input())
balloons = list(map(int, input().split()))
target = 0
answer = []
idx = [x for x in range(1,n + 1)]
tmp = balloons.pop(target)
answer.append(idx.pop(target))

while balloons:
    if tmp < 0:
        target = (target + tmp) % len(balloons)
    else:
        target = (target + tmp - 1) % len(balloons)
    tmp = balloons.pop(target)
    answer.append(idx.pop(target))

for i in answer:
    print(i, end=' ')

File: Baekjoon/구현/2346.cpp
// https://www.acmicpc.net/problem/2346

#include <iostream>
#include <vector>

using namespace std;

int main() {
    int n;
    cin >> n;

    vector<int> balloons(n);
    vector<int> idx(n);
    vector<int> answer;

    for (int i = 0; i < n; i++) {
        cin >> balloons[i];
        idx[i] = i + 1;
    }

    int target = 0;
    int tmp = balloons[target];
    answer.push_back(idx[target]);
    balloons.erase(balloons.begin() + target);
    idx.erase(idx.begin() + target);

    while (!balloons.empty()) {
        if (tmp < 0) {
            target = (target + tmp) % static_cast<int>(balloons.size());
            if (target < 0) {
                target += balloons.size();
            }
        } else {
            target = (target + tmp - 1) % static_cast<int>(balloons.size());
        }

        tmp = balloons[target];
        answer.push_back(idx[target]);
        balloons.erase(balloons.begin() + target);
        idx.erase(idx.begin() + target);
    }

    for (int num : answer) {
        cout << num << " ";
    }
    cout << endl;

    return 0;
}